tests/Moq.Tests/EventHandlersFixture.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2007, Clarius Consulting, Manas Technology Solutions, InSTEDD, and Contributors.
+// All rights reserved. Licensed under the BSD 3-Clause License; see License.txt.
+
+using System;
+
+using Xunit;
+
+namespace Moq.Tests
+{
+	public class EventHandlersFixture
+	{
+		[Fact]
+		public void Raising_event__directly_on_mock_object__does_not_trigger_handler__if_CallBase_false()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.False(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__triggers_handler__if_CallBase_true()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>() { CallBase = true };
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.True(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__does_not_trigger_handler__if_setup_without_CallBase_present()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.SetupAdd(m => m.Event += It.IsAny<Action>());
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.False(handled);
+		}
+
+		[Fact]
+		public void Raising_event__directly_on_mock_object__triggers_handler__if_setup_with_CallBase_present()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.SetupAdd(m => m.Event += It.IsAny<Action>()).CallBase();
+			mock.Object.Event += () => handled = true;
+			mock.Object.RaiseEvent();
+			Assert.True(handled);
+		}
+
+		[Fact]
+		public void Raising_event__using_Raise__triggers_handler__if_CallBase_false()
+		{
+			var handled = false;
+			var mock = new Mock<HasEvent>();
+			mock.Object.Event += () => handled = true;
+			mock.Raise(m => m.Event += null);
+			Assert.True(handled);
+		}
+
+		[Fact]
+		public void Raising_event__using_Raise__does_not_trigger_handler__if_CallBase_true()
+		{
```

Comment:
... and this test is also not clear to my why CallBase makes a difference. 
I stop now annotating the tests, as it seems that there is some fundamental difference with CallBase = {true, false} I don't understand.

Reply:
Continuing from my explanation above, this test demonstrates how things go wrong when you use `.CallBase()` as well as `Mock.Raise`: the event handlers won't trigger.